crates/partition-store/src/outbox_table/mod.rs

```diff
@@ -39,6 +39,33 @@ fn add_message<S: StorageAccess>(
     storage.put_kv(key, outbox_message);
 }
 
+fn get_outbox_head_seq_number<S: StorageAccess>(
+    storage: &mut S,
+    partition_id: PartitionId,
+) -> Result<Option<u64>> {
+    let _x = RocksDbPerfGuard::new("get-head-outbox");
+    let start = OutboxKey::default().partition_id(partition_id);
+
+    let end = OutboxKey::default()
+        .partition_id(partition_id)
+        .message_index(u64::MAX);
+
+    let mut iter = storage.iterator_from(TableScan::KeyRangeInclusiveInSinglePartition(
+        partition_id,
+        start,
+        end,
+    ));
+
+    iter.seek_to_first();
+    if iter.valid() {
+        let k = iter.key().unwrap();
+        let key = OutboxKey::deserialize_from(&mut Cursor::new(k))?;
+        Ok(key.message_index)
+    } else {
+        Ok(None)
+    }
```

Comment:
I also tried the following approach (copied from `get_next_outbox_message`):

```rs
storage.get_first_blocking(
    TableScan::KeyRangeInclusiveInSinglePartition(partition_id, start, end),
    |kv| {
        if let Some((k, _)) = kv {
            let t = decode_key_value(k, &[])?;
            Ok(Some(t.0))
        } else {
            Ok(None)
        }
    },
)
```

But it failed with:

```
--- STDERR:              restate-partition-store::integration_test test_read_write ---
!!!! Runnning with test-util enabled !!!!
thread 'test_read_write' panicked at crates/partition-store/tests/outbox_table_test/mod.rs:43:10:
should not fail: Conversion(failed reading codec: remaining bytes in buf '0' < version bytes '1'

Stack backtrace:
   0: std::backtrace_rs::backtrace::libunwind::trace
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/../../backtrace/src/backtrace/libunwind.rs:105:5
   1: std::backtrace_rs::backtrace::trace_unsynchronized
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2: std::backtrace::Backtrace::create
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/backtrace.rs:331:13
   3: anyhow::error::<impl core::convert::From<E> for anyhow::Error>::from
             at /Users/pavel/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.81/src/backtrace.rs:27:14
   4: <T as core::convert::Into<U>>::into
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/convert/mod.rs:759:9
   5: restate_partition_store::outbox_table::decode_value::{{closure}}
             at ./src/outbox_table/mod.rs:229:51
   6: core::result::Result<T,E>::map_err
             at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/result.rs:829:27
   7: restate_partition_store::outbox_table::decode_value
             at ./src/outbox_table/mod.rs:228:26
   8: restate_partition_store::outbox_table::decode_key_value
             at ./src/outbox_table/mod.rs:221:26
   9: restate_partition_store::outbox_table::get_outbox_head_seq_number::{{closure}}
             at ./src/outbox_table/mod.rs:73:25
  10: restate_partition_store::partition_store::StorageAccess::get_first_blocking
             at ./src/partition_store.rs:686:9
  11: restate_partition_store::outbox_table::get_outbox_head_seq_number
             at ./src/outbox_table/mod.rs:69:5
  12: restate_partition_store::outbox_table::<impl restate_storage_api::outbox_table::ReadOnlyOutboxTable for restate_partition_store::partition_store::RocksDBTransaction>::get_outbox_head_seq_number::{{closure}}
             at ./src/outbox_table/mod.rs:180:9
  13: integration_test::outbox_table_test::verify_outbox_head_seq_number::{{closure}}
             at ./tests/outbox_table_test/mod.rs:42:10
  14: integration_test::outbox_table_test::run_tests::{{closure}}
             at ./tests/outbox_table_test/mod.rs:102:54
```

I didn't spend too much time trying to work out why that is but curious if you have any insights into what's going on here?

Reply:
Because you are passing `&[]` as the value. If you replace `decode_key_value` with `OutboxKey::deserialize_from(&mut Cursor::new(k))?` it should probably work.